scripts/prepack.ts

```diff
@@ -22,7 +22,12 @@ export async function preparePackage({ sourcePackageDir }: { sourcePackageDir: s
 
 	copyFileSync(
 		path.join(sourcePackageDir, `api/public.d.ts`),
-		path.join(sourcePackageDir, 'index.d.ts')
+		path.join(sourcePackageDir, 'dist/cjs/index.d.ts')
+	)
+
+	copyFileSync(
+		path.join(sourcePackageDir, `api/public.d.ts`),
+		path.join(sourcePackageDir, 'dist/esm/index.d.ts')
```

Comment:
is it necessary to have the typings duplicated in both dist folders?

Reply:
Not necessarily, I was originally going off of the Twitter thread:
```
5. (cont) In short, you cannot use a single *.d.ts file to type an ESM and CJS implementation file at the same time. Another way of saying the same thing is that the declaration file extension has to match the extension of what it’s typing—.d.ts=.js, .d.cjs = .cjs, etc...
```

But then the api extractor etc. blew up in my face, but I left them in the separate folders in case this points needs to be picked up later. I should have clarified in the comments.

Full thread for reference
 https://twitter.com/atcb/status/1634653474041503744